pkg/sdkserver/sdkserver.go

```diff
@@ -985,41 +999,238 @@ func (s *SDKServer) GetList(ctx context.Context, in *alpha.GetListRequest) (*alp
 	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
 		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
 	}
-	// TODO(#2716): Implement me
-	return nil, errors.Errorf("Unimplemented -- GetList coming soon")
+	s.logger.WithField("name", in.Name).Debug("Getting List")
+
+	gs, err := s.gameServer()
+	if err != nil {
+		return nil, err
+	}
+
+	s.gsUpdateMutex.RLock()
+	defer s.gsUpdateMutex.RUnlock()
+
+	list, ok := gs.Status.Lists[in.Name]
+	if !ok {
+		return nil, errors.Errorf("list not found: %s", in.Name)
+	}
+
+	s.logger.WithField("Get List", list).Debugf("Got List %s", in.Name)
+	protoList := alpha.List{Name: in.Name, Values: list.Values, Capacity: list.Capacity}
+	// If there are batched changes that have not yet been applied, apply them to the List.
+	// This does NOT validate batched the changes.
+	if listUpdate, ok := s.gsListUpdates[in.Name]; ok {
+		if listUpdate.capacitySet != nil {
+			protoList.Capacity = *listUpdate.capacitySet
+		}
+		// TODO: DELETE
+		// if len(listUpdate.listDelete) != 0 {
+		// }
+		// TODO: APPEND
+		// if len(listUpdate.listAppend) != 0 {
+		// }
+		// TODO: Truncate if past capacity
+		s.logger.WithField("Get List", list).Debugf("Applied Batched List Updates %v", listUpdate)
+	}
+
+	return &protoList, nil
 }
 
-// UpdateList returns the updated List.
+// UpdateList collapses all update capacity requests for a given List into a single UpdateList request.
+// This function currently only updates the Capacity of a List.
+// Returns error if the List does not exist (name cannot be updated).
+// Returns error if the List update capacity is out of range [0,1000].
 // [Stage:Alpha]
 // [FeatureFlag:CountsAndLists]
 func (s *SDKServer) UpdateList(ctx context.Context, in *alpha.UpdateListRequest) (*alpha.List, error) {
 	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
 		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
 	}
-	// TODO(#2716): Implement Me
-	return nil, errors.Errorf("Unimplemented -- UpdateList coming soon")
+
+	if in.List == nil || in.UpdateMask == nil {
+		return nil, errors.Errorf("INVALID_ARGUMENT. List: %v and UpdateMask %v cannot be nil", in.List, in.UpdateMask)
```

Comment:
Nit: I'm guessing this code is from before we moved to standard Go errors, but just in case - something like:

```suggestion
		return nil, errors.Errorf("invaid argument. List: %v and UpdateMask %v cannot be nil", in.List, in.UpdateMask)
```